postgres/datadog_checks/postgres/postgres.py

```diff
@@ -66,14 +48,12 @@ def __init__(self, name, init_config, instances):
                 "rather than the now deprecated custom_metrics"
             )
         self.config = PostgresConfig(self.instance)
+        self.metric_utils = PostgresMetrics(self.config)
```

Comment:
Seems a bit strange to have an instance of `metric_utils`.

What about `PostgresMetrics` => `MetricsCache` ? Same for the variables and file name (metrics_cache.py).